### ISSUE TITLE
require-returns-check has false negative when a return statement is present in nested function

### ISSUE THREAD

**Issue by golopot** [gajus/eslint-plugin-jsdoc#173]
### code
```javascript
/**
 * This should error because
 * foo does not explicitly return but has a returns tag 
 * @returns .
 */
function foo() {
  let a = () => {
    return 5;
  }
}
```
### expected
reports error

### actual
no error

### why

This happens because the current determination of presence of  `return` statement is done by a string search in the function source code.

**Comment by golopot** [comment]
Close as duplicate of #151 .
